tasks/pipeline.py

```diff
@@ -160,6 +161,37 @@ def trigger(
     )
 
 
+@task
+def auto_cancel_previous_pipelines(ctx):
+    """
+    Automatically cancel previous pipelines running on the same ref
+    """
+
+    project_name = "DataDog/datadog-agent"
+    if not os.environ.get('GITLAB_TOKEN'):
+        raise Exit("GITLAB_TOKEN variable needed to cancel pipelines on the same ref.", 1)
+
+    gitlab = Gitlab(project_name=project_name, api_token=get_gitlab_token())
+    gitlab.test_project_found()
+
+    git_ref = os.getenv("CI_COMMIT_REF_NAME")
+    git_sha = os.getenv("CI_COMMIT_SHA")
+
+    pipelines = get_running_pipelines_on_same_ref(gitlab, git_ref)
+    pipelines_without_current = [p for p in pipelines if p["sha"] != git_sha]
+
+    for pipeline in pipelines_without_current:
+        # We cancel pipeline only if it correspond to a commit that is an ancestor of the current commit
+        try:
+            ctx.run(f'git merge-base --is-ancestor {pipeline["sha"]} {git_sha}')
```

Comment:
Could you clarify the intended logic? 
What happens in a 1 commit PR? Wouldn't the common ancestor be `main` in this case?

Reply:
Oh OK then I completely misunderstood the test, thanks for the explanation, this makes a lot more sense to me now :)